### PR TITLE
UICIRC-825: Add RTL/Jest testing for  component in

### PR DESCRIPTION
**Pull request by pavelspichonak**
## Purpose
Add RTL/Jest testing for `RulesForm` component in `src/settings/lib/RuleEditor`.

## Refs
https://issues.folio.org/browse/UICIRC-825

## Screenshots
Screen Shot 2022-06-21 at 5 33 29 PM

### CODE REVIEW

File: src/settings/lib/RuleEditor/RulesForm.test.js [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+  describe('with default props', () => {
+    beforeEach(() => {
+      render(
+        <RulesForm {...testDefaultProps} />
+      );
+    });
+
+    it('should call submit callback when form is submitted', () => {
+      fireEvent.submit(screen.getByTestId(testIds.formLoanRules));
+      expect(testDefaultProps.handleSubmit).toHaveBeenCalled();
```
Comment (Stanislau-Snapok) [inline src/settings/lib/RuleEditor/RulesForm.test.js:58]: empty line before expect

File: src/settings/lib/RuleEditor/RulesForm.test.js [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+      }), {});
+    });
+  });
+
+  describe('filterRules method', () => {
+    const newRuleFilterValue = 'newRuleFilterValue';
+
+    beforeEach(() => {
+      TextField.mockImplementationOnce(({ onChange }) => (
+        <div data-testid={testIds.ruleFilter} onClick={() => onChange({
```
Comment (Stanislau-Snapok) [inline src/settings/lib/RuleEditor/RulesForm.test.js:129]: check please eslint errors

**Review by Stanislau-Snapok — APPROVED** [review]

File: src/settings/lib/RuleEditor/RulesForm.test.js [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+        <RulesForm {...testDefaultProps} />
+      );
+    });
+
+    it('should call submit callback when form is submitted', () => {
+      fireEvent.submit(screen.getByTestId(testIds.formLoanRules));
+      expect(testDefaultProps.handleSubmit).toHaveBeenCalled();
+    });
+
+    it('should render TextField component', () => {
```
Comment (pavelspichonak) [inline src/settings/lib/RuleEditor/RulesForm.test.js:61]: => with correct props

File: src/settings/lib/RuleEditor/RulesForm.test.js [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+      expect(TextField).toHaveBeenCalledWith(expect.objectContaining({
+        'aria-label': labelIds.filterRules,
+        value: '',
+        validationEnabled: false,
+        placeholder: labelIds.filterRules,
+      }), {});
+    });
+
+    it('should render Button component', () => {
```
Comment (pavelspichonak) [inline src/settings/lib/RuleEditor/RulesForm.test.js:70]: => with correct props

File: src/settings/lib/RuleEditor/RulesForm.test.js [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+      expect(Button).toHaveBeenCalledWith(expect.objectContaining({
+        fullWidth: true,
+        id: 'clickable-save-loan-rules',
+        type: 'submit',
+        disabled: testDefaultProps.pristine || testDefaultProps.submitting,
+        children: labelIds.checkoutSave,
+      }), {});
+    });
+
+    it('should render Field component', () => {
```
Comment (pavelspichonak) [inline src/settings/lib/RuleEditor/RulesForm.test.js:80]: => with correct props

File: src/settings/lib/RuleEditor/RulesForm.test.js [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+    beforeEach(() => {
+      render(
+        <RulesForm
+          {...testDefaultProps}
+          pristine
+        />
+      );
+    });
+
+    it('should render Button component', () => {
```
Comment (pavelspichonak) [inline src/settings/lib/RuleEditor/RulesForm.test.js:100]: should render disabled button

File: src/settings/lib/RuleEditor/RulesForm.test.js [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+      render(
+        <RulesForm
+          {...testDefaultProps}
+          submitting
+        />
+      );
+    });
+
+    it('should render Button component', () => {
+      expect(Button).toHaveBeenCalledWith(expect.objectContaining({
```
Comment (pavelspichonak) [inline src/settings/lib/RuleEditor/RulesForm.test.js:118]: should render disabled button

File: src/settings/lib/RuleEditor/RulesForm.test.js [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+          },
+        })} />
+      ));
+
+      render(
+        <RulesForm {...testDefaultProps} />
+      );
+    });
+
+    it('should render Field component', () => {
```
Comment (pavelspichonak) [inline src/settings/lib/RuleEditor/RulesForm.test.js:141]: => with changed filter prop value

**Review by Dmitriy-Litvinenko — APPROVED** [review]

File: src/settings/lib/RuleEditor/RulesForm.test.js [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+        disabled: true,
+      }), {});
+    });
+  });
+
+  describe('filterRules method', () => {
+    const newRuleFilterValue = 'newRuleFilterValue';
+
+    beforeEach(() => {
+      TextField.mockImplementationOnce(({ onChange }) => (
```
Comment (pavelspichonak) [inline src/settings/lib/RuleEditor/RulesForm.test.js:128]: fix linter errors

File: src/settings/lib/RuleEditor/RulesForm.test.js (hunk 1 above)
Comment (pavelspichonak) [reply src/settings/lib/RuleEditor/RulesForm.test.js:59]: done

File: src/settings/lib/RuleEditor/RulesForm.test.js (hunk 3 above)
Comment (pavelspichonak) [reply src/settings/lib/RuleEditor/RulesForm.test.js]: done

File: src/settings/lib/RuleEditor/RulesForm.test.js (hunk 4 above)
Comment (pavelspichonak) [reply src/settings/lib/RuleEditor/RulesForm.test.js]: done

File: src/settings/lib/RuleEditor/RulesForm.test.js (hunk 6 above)
Comment (pavelspichonak) [reply src/settings/lib/RuleEditor/RulesForm.test.js]: done

File: src/settings/lib/RuleEditor/RulesForm.test.js (hunk 7 above)
Comment (pavelspichonak) [reply src/settings/lib/RuleEditor/RulesForm.test.js:119]: done

File: src/settings/lib/RuleEditor/RulesForm.test.js (hunk 8 above)
Comment (pavelspichonak) [reply src/settings/lib/RuleEditor/RulesForm.test.js]: done

File: src/settings/lib/RuleEditor/RulesForm.test.js (hunk 2 above)
Comment (pavelspichonak) [reply src/settings/lib/RuleEditor/RulesForm.test.js]: done

**Pull request merged**